[ENG-11094][docs] improve local builds docs (#26197)
expo/expo: docs/pages/build-reference/local-builds.mdx

import { Terminal } from '~/ui/components/Snippet';

You can run the same build process that we run on the EAS Build servers directly on your machine with the `--local` flag which is a useful way to debug build failures happening on your EAS cloud builds.

> > **Note:** To make builds using the Expo framework locally use [`npx expo run:android` or `npx expo run:ios` commands](/more/expo-cli/#compiling) instead. The `eas build --local` command is a way to make builds using [EAS cloud services](/build/introduction) locally. If you use [Continuous Native Generation](/workflow/continuous-native-generation), you can also run [prebuild](/workflow/prebuild) to generate your **android** and **ios** directories and then proceed to open the projects in the respective IDEs. Further, you can build them just like any native project.

Reviewer: ```suggestion
> **Note:** The `eas build --local` command is a way to make builds using [EAS cloud services](/build/introduction) on your local machine, following the same process that would otherwise run remotely on our cloud. To compile and run your app locally with Expo CLI, use [`npx expo run:android` or `npx expo run:ios` commands](/more/expo-cli/#compiling) instead.  If you use [Continuous Native Generation](/workflow/continuous-native-generation), you can also run [prebuild](/workflow/prebuild) to generate your **android** and **ios** directories and then proceed to open the projects in the respective IDEs. Further, you can build them just like any native project.
```

> > **Note:** The `eas build --local` command is a way to make builds using [EAS cloud services](/build/introduction) on your local machine, following the same process that would otherwise run remotely on our cloud. To compile and run your app locally with Expo CLI, use [`npx expo run:android` or `npx expo run:ios` commands](/more/expo-cli/#compiling) instead.  If you use [Continuous Native Generation](/workflow/continuous-native-generation), you can also run [prebuild](/workflow/prebuild) to generate your **android** and **ios** directories and then proceed to open the projects in the respective IDEs. Further, you can build them just like any native project.

Reviewer: ```suggestion
> **Note:** To compile and run your app locally with Expo CLI, use [`npx expo run:android` or `npx expo run:ios` commands](/more/expo-cli/#compiling) instead. If you use [Continuous Native Generation](/workflow/continuous-native-generation), you can also run [prebuild](/workflow/prebuild) to generate your **android** and **ios** directories and then proceed to open the projects in the respective IDEs and build them like any native project. With both of these approaches, you will not be following the exact process that is run on the cloud with EAS Build — that is what the `eas build --local` flag is for.
```

import { Terminal } from '~/ui/components/Snippet';

You can run the same build process that we run on the EAS Build servers directly on your machine with the `--local` flag which is a useful way to debug build failures happening on your EAS cloud builds.

> > **Note:** For making builds using the Expo framework locally use `npx expo run:android` or `npx expo run:ios` [commands](/more/expo-cli/#compiling) instead. The `eas build --local` command is a way to make builds using [EAS cloud services](/build/introduction) locally.

Reviewer 2: ```suggestion
> **Note:** To make builds using the Expo framework locally, use `[npx expo run:android` or `npx expo run:ios` commands](/more/expo-cli/#compiling) instead. The `eas build --local` command is a way to make builds using [EAS cloud services](/build/introduction) locally.
```

Let's highlight the whole set of commands with the link.

## Use cases for local builds

> - [Debugging](#using-local-builds-for-debugging) build failures on EAS servers.

Reviewer 2: ```suggestion
- [Debugging](#use-local-builds-for-debugging) build failures on EAS servers.
```

Fix the internal doc link.

- `EAS_LOCAL_BUILD_WORKINGDIR` - Specify the working directory for the build process, by default it's somewhere (it's platform dependent) in **/tmp** directory.
- `EAS_LOCAL_BUILD_ARTIFACTS_DIR` - The directory where artifacts are copied after a successful build. By default these files are copied to the current directory, which may be undesirable if you are running many consecutive builds.

> If you use `EAS_LOCAL_BUILD_SKIP_CLEANUP` and `EAS_LOCAL_BUILD_WORKINGDIR` for iOS builds you should be able to inspect the contents of the `logs` subdirectory of the working directory to take a look at your Xcode logs.

Reviewer: ```suggestion
If you use `EAS_LOCAL_BUILD_SKIP_CLEANUP` and `EAS_LOCAL_BUILD_WORKINGDIR` for iOS builds you should be able to inspect the contents of the `logs` subdirectory of the working directory to read your Xcode logs.
```

  - fastlane (iOS only)
  - CocoaPods (iOS only)
  - Android SDK and NDK

> - On Windows, you can use [WSL](https://docs.microsoft.com/en-us/windows/wsl/install) for local EAS Builds as it seems to work well. However, we do not officially test it and do not support Windows. Only macOS and Linux are supported.

Reviewer: ```suggestion
- On Windows, you can use [WSL](https://docs.microsoft.com/en-us/windows/wsl/install) for local EAS Builds — however, we do not officially test against this platform and do not support Windows for local builds (macOS and Linux are supported).
```

  - fastlane (iOS only)
  - CocoaPods (iOS only)
  - Android SDK and NDK

> - Windows is not supported (only macOS and Linux are supported). The suggested way to run local EAS builds on Windows is to use [WSL](https://docs.microsoft.com/en-us/windows/wsl/install) even though it's also not officially supported (we do not test it), but it seems to work well.

Reviewer 2: ```suggestion
- On Windows, you can use [WSL](https://docs.microsoft.com/en-us/windows/wsl/install) for local EAS Builds as it seems to work well. However, we do not officially test it and do not support Windows. Only macOS and Linux are supported.
```

Let's be a bit concise here and offer our message with clarity.
